### PR TITLE
Implement Suicide Implants

### PR DESCRIPTION
**Pull request by HammerGS**
Implements Explosive Suicide Implants as per Interstellar Operations pg 83. This MD (Manei Domini) enhancement allows infantry, Battle Armor, MekWarriors, pilots, and vehicle crews to detonate implanted explosives.

  Features

  Voluntary Detonation (Firing Phase)

  - New "Suicide Implants" button in Firing Display when unit has the ability
  - Dialog allows selecting number of troopers to detonate (infantry/BA)
  - Confirmation dialog for Mek/Aero/Vehicle pilots

  Damage Effects by Entity Type

  | Entity Type           | Effect                                                         |
  |-----------------------|----------------------------------------------------------------|
  | Conventional Infantry | 0.57 damage per trooper to ALL units in hex                    |
  | Battle Armor          | Destroys selected troopers only, no area damage                |
  | Mek                   | 1 IS to head + crit roll + cockpit destroyed + pilot killed    |
  | Aero                  | 1 armor to nose + crit roll + cockpit destroyed + pilot killed |
  | Vehicle               | Crew Killed + 1 IS to all facings + crit rolls                 |

  Reactive Detonation

  When conventional infantry with suicide implants loses troopers to enemy fire, the dead troopers automatically detonate dealing 0.57 damage per trooper to all enemies in the same hex.

  Additional Effects

  - Building damage: CF reduced by (troopers / 2) when detonating inside structures
  - BV bonus: +0.12 per trooper for conventional infantry only

  Rules Reference

  - Interstellar Operations pg 83

  Files Changed

  New Files

  - SuicideImplantsAttackAction.java - Attack action with damage calculations
  - SuicideImplantsDialog.java - UI dialog for trooper selection
  - SuicideImplantsTest.java - Comprehensive unit tests

  Modified Files

  - TWGameManager.java - Resolution methods for all entity types
  - TWDamageManager.java - Reactive detonation on trooper death
  - FiringDisplay.java - Button and command handling
  - DamageType.java - Added SUICIDE_IMPLANT_REACTION to prevent recursion
  - InfantryBVCalculator.java - BV bonus calculation
  - messages.properties / report-messages.properties - UI strings and reports

  Unrelated Cleanup (bundled)

  - 267 weapon files: rulesRefs format standardization (e.g., "90, IO:AE" → "96, IO")

  Test Plan

  ### Conventional Infantry Tests
  - [x] Infantry with implants in same hex as enemy - Button available, damage to all enemies in hex
  - [x] 21 troopers detonate - 12 damage (21 × 0.57 = 11.97 → 12)
  - [x] 10 troopers detonate - 6 damage (10 × 0.57 = 5.7 → 6)
  - [x] Slider allows partial detonation - Only selected troopers die
  - [x] Infantry in building - Building takes CF damage = troopers/2
  - [x] Multiple enemies in hex - All enemies take damage

  ### Reactive Detonation Tests
  - [x] Enemy kills 3 troopers - 2 damage to attacker (if in same hex)
  - [x] Enemy kills 6 troopers - 3 damage to attacker (if in same hex)
  - [x] Attacker in different hex - No reactive damage (correct behavior)
  - [x] Friendly fire (same team) - No reactive damage (correct behavior)
  - [x] Multiple enemies in hex when troopers die - All enemies take reactive damage

  ### Battle Armor Tests
  - [x] 4-trooper BA, detonate 2 - 2 troopers destroyed, 2 remain, NO area damage
  - [x] BA detonates all troopers - Squad destroyed
  - [x] BA taking damage - NO reactive detonation (BA don't trigger reactive)

  ### Mek Pilot Tests
  - [x] MekWarrior detonates - 1 IS damage to head
  - [x] Critical hit roll performed on head
  - [x] Cockpit marked destroyed (for salvage)
  - [x] Pilot killed

  ### Aero Pilot Tests
  - [x] Aero pilot detonates - 1 armor damage to nose
  - [x] Critical hit roll performed
  - [x] Pilot killed

  ### Vehicle Crew Tests
  - [x] Vehicle crew detonates - Crew Killed critical
  - [x] Internal damage - 1 IS to ALL facings
  - [x] Critical rolls for each facing with structure

  ### Edge Cases
  - [x] Unconscious crew - Cannot detonate (IMPOSSIBLE)
  - [x] Dead crew - Cannot detonate (IMPOSSIBLE)
  - [x] Being transported - Cannot detonate (IMPOSSIBLE)
  - [x] Unit without Suicide Implants ability - Button not available
  - [x] BV calculation - Infantry with implants has higher BV than without

### CODE REVIEW

File: megamek/src/megamek/server/totalWarfare/TWGameManager.java [hunk 1]
```diff
@@ -13010,6 +13012,328 @@ private void resolveToxinAttack(PhysicalResult physicalResult, int lastEntityId)
...
+        } else if (attackingEntity instanceof Mek mek) {
+            resolveMekPilotSuicide(mek);
+        } else if (attackingEntity instanceof Aero aero) {
+            resolveAeroPilotSuicide(aero);
+        } else if (attackingEntity instanceof Tank tank) {
+            resolveVehicleCrewSuicide(tank);
+        }
+
+        addNewLines();
+    }
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/server/totalWarfare/TWGameManager.java:13119]: Addressed here https://github.com/MegaMek/megamek/pull/7837/commits/1630231d6dde1d1b122045989c24c824d0715e32

File: megamek/src/megamek/server/totalWarfare/TWGameManager.java [hunk 2]
```diff
@@ -13010,6 +13012,328 @@ private void resolveToxinAttack(PhysicalResult physicalResult, int lastEntityId)
         addNewLines();
     }
 
+    /**
+     * Handle a Suicide Implants attack (IO pg 83). The entity detonates their explosive implants, destroying themselves
+     * and causing damage based on entity type: - Conventional Infantry: 0.57 damage per trooper to all entities in hex
+     * + building CF damage - Battle Armor: Destroys selected troopers only, no damage to others - MekWarrior/Pilot: 1
+     * IS to Mek head (or 1 armor to fighter nose) + crit + cockpit destroyed - Vehicle Crew: Crew Killed + 1 IS to all
+     * facings + crit rolls
+     */
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/server/totalWarfare/TWGameManager.java]: Addressed here https://github.com/MegaMek/megamek/pull/7837/commits/1630231d6dde1d1b122045989c24c824d0715e32

File: megamek/unittests/megamek/common/SuicideImplantsTest.java [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+            Infantry withImplants = createInfantry(21, true, 0);
+            Infantry withoutImplants = createInfantry(21, false, 0);
+
+            int bvWith = withImplants.calculateBattleValue();
+            int bvWithout = withoutImplants.calculateBattleValue();
+
+            assertTrue(bvWith > bvWithout,
+                  "Infantry with Suicide Implants should have higher BV. " +
+                        "With: " + bvWith + ", Without: " + bvWithout);
+        }
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/unittests/megamek/common/SuicideImplantsTest.java]: Addressed here https://github.com/MegaMek/megamek/pull/7837/commits/1630231d6dde1d1b122045989c24c824d0715e32

File: megamek/src/megamek/server/totalWarfare/TWGameManager.java [hunk 4]
```diff
@@ -13010,6 +13012,328 @@ private void resolveToxinAttack(PhysicalResult physicalResult, int lastEntityId)
...
+            // Destroy individual troopers starting from the highest numbered location
+            for (int i = 0; i < troopersDetonating; i++) {
+                for (int loc = battleArmor.locations() - 1; loc >= BattleArmor.LOC_TROOPER_1; loc--) {
+                    if (battleArmor.getInternal(loc) > 0) {
+                        battleArmor.setInternal(0, loc);
+                        battleArmor.setArmor(0, loc);
+                        break;
+                    }
+                }
+            }
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/server/totalWarfare/TWGameManager.java:13217]: Addressed here https://github.com/MegaMek/megamek/pull/7837/commits/1630231d6dde1d1b122045989c24c824d0715e32

File: megamek/resources/megamek/common/report-messages.properties [hunk 5]
```diff
@@ -676,6 +676,24 @@
...
+4587=The pilot's implants detonate inside the cockpit!
+4588=The Mek takes 1 internal damage to the head.
+4589=Checking for critical hits...
+4590=The pilot is killed by the explosion.
+4591=The pilot's implants detonate inside the cockpit!
+4592=The cockpit is destroyed.
+4593=The crew's implants detonate inside the vehicle!
+4594=The crew is killed!
+4595=1 internal damage to <data>.
+4596=<data> dead troopers trigger reactive suicide implant detonation! <data> damage to all enemies in hex!
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/resources/megamek/common/report-messages.properties:699]: Addressed here https://github.com/MegaMek/megamek/pull/7837/commits/1630231d6dde1d1b122045989c24c824d0715e32

File: megamek/src/megamek/server/totalWarfare/TWGameManager.java [hunk 6]
```diff
@@ -13010,6 +13012,328 @@ private void resolveToxinAttack(PhysicalResult physicalResult, int lastEntityId)
...
+            report.subject = target.getId();
+            report.indent(2);
+            report.addDesc(target);
+            report.add(damage);
+            addReport(report);
+
+            HitData hit = target.rollHitLocation(ToHitData.HIT_NORMAL, ToHitData.SIDE_FRONT);
+            hit.setGeneralDamageType(HitData.DAMAGE_PHYSICAL);
+            addReport(damageEntity(target, hit, damage));
+        }
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/server/totalWarfare/TWGameManager.java:13151]: Addressed here https://github.com/MegaMek/megamek/pull/7837/commits/1630231d6dde1d1b122045989c24c824d0715e32

File: megamek/src/megamek/server/totalWarfare/TWDamageManager.java [hunk 7]
```diff
@@ -302,6 +304,18 @@ public Vector<Report> damageEntity(Entity entity, HitData hit, int damage, boole
         boolean tookAnyDamage = damage > 0; // Track if any damage was applied for Proto DNI feedback
         Hex te_hex = null;
 
+        // Track initial trooper count for suicide implant reactive damage (IO pg 83)
+        // Only track if: conventional infantry, has suicide implants, and this is NOT reactive damage
+        int initialTroopers = -1;
+        boolean hasSuicideImplants = entity.hasAbility(OptionsConstants.MD_SUICIDE_IMPLANTS);
+        boolean checkSuicideImplantReaction = isPlatoon
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/server/totalWarfare/TWDamageManager.java:310]: Field gun platoons are conventional infantry. Not valid.

File: megamek/resources/megamek/common/report-messages.properties [hunk 8]
```diff
@@ -676,6 +676,24 @@
 4561=, but the toxin attack is impossible (<data>).
 4562=<span class='warning'><B>misses</B></span>; the noxious cloud disperses harmlessly.
 4563=<span class='success'><B>hits</B></span>! <data> chokes and gasps as the toxin burns through their lungs, taking <data> damage.
+# Suicide Implants Attack (IO pg 83)
+4580=<data> detonates their explosive suicide implants!
+4581=, but detonation is impossible (<data>).
+4582=The explosion deals <data> damage to all units in the hex!
+4583=<data> takes <data> damage from the explosion.
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/resources/megamek/common/report-messages.properties:686]: Valid concern, but this affects the entire report system (every <data> tag with entity names). Out of scope for this PR - would need a broader security review of the Report class

**Comment by HammerGS** [comment]
I've retested everything no major issues but need clean up a few minor things.

File: megamek/src/megamek/client/ui/dialogs/phaseDisplay/SuicideImplantsDialog.java [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+import megamek.common.units.Entity;
+import megamek.common.units.Mek;
+import megamek.common.units.Tank;
+
+/**
+ * Dialog for configuring suicide implant detonation.
+ * <p>
+ * For conventional infantry and Battle Armor, this dialog presents a slider to select the number of troopers to
+ * detonate, along with a damage preview.
+ * <p>
```
Comment (IllianiBird) [inline megamek/src/megamek/client/ui/dialogs/phaseDisplay/SuicideImplantsDialog.java:67]: Unpaired `<p>` tags. This isn't a problem, just an FYI. Sometimes code style will throw a wobbly with these and fail your PR. Ideally, you want to always close any tags you open. i.e. `<p>` should be paired with `</p>`

File: megamek/src/megamek/client/ui/dialogs/phaseDisplay/SuicideImplantsDialog.java [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+
+        mainPanel.add(javax.swing.Box.createVerticalStrut(UIUtil.scaleForGUI(5)));
+
+        // Slider panel
+        JPanel sliderPanel = new JPanel(new FlowLayout(FlowLayout.CENTER));
+
+        JLabel minLabel = new JLabel("1");
+        trooperSlider = new JSlider(SwingConstants.HORIZONTAL, 1, maxTroopers, maxTroopers);
+        trooperSlider.addChangeListener(this);
+        trooperSlider.setMajorTickSpacing(Math.max(1, maxTroopers / 5));
```
Comment (IllianiBird) [inline megamek/src/megamek/client/ui/dialogs/phaseDisplay/SuicideImplantsDialog.java]: Note that this will use integer division, that means both 0.1 and 0.999999 will return 0. Essentially it's 'round down'. Can I ask you to verify that this is the intended behavior?

Ideally, but not required, you'd use explicit division:

```
int tickSpacing = (int) floor(maxTroopers / 5);
trooperSlider.setMajorTickSpacing(Math.max(1, tickSpacing);
```

This avoid a potential future event where another developer sees that we're using integer division and 'fixes' it.

File: megamek/src/megamek/client/ui/panels/phaseDisplay/FiringDisplay.java [hunk 11]
```diff
@@ -2059,6 +2064,40 @@ private void updateActivateSPA() {
...
+    private void updateSuicideImplants() {
+        Entity entity = currentEntity();
+        boolean canDetonate = (entity != null)
+              && entity.hasAbility(OptionsConstants.MD_SUICIDE_IMPLANTS)
+              && entity.isActive()
+              && (entity.getCrew() != null)
+              && !entity.getCrew().isDead()
+              && !entity.getCrew().isUnconscious()
+              && (entity.getTransportId() == Entity.NONE);
+        setSuicideImplantsEnabled(canDetonate);
```
Comment (IllianiBird) [inline megamek/src/megamek/client/ui/panels/phaseDisplay/FiringDisplay.java]: I'd really prefer this to be broken up into several boolean variables and have those variables passed in. I've mentioned this before, but it basically makes chained conditionals _much_ easier to read.

For example

```
boolean exampleA = entity.hasAbility(OptionsConstants.MD_SUICIDE_IMPLANTS);
boolean exampleB = entity.isActive();
boolean exampleC = entity.getCrew() != null;
boolean canDetonate = entity != null &&
    exampleA &&
    exampleB &&
    exampleC;
```

Vs.

```
boolean canDetonate = entity != null &&
    entity.hasAbility(OptionsConstants.MD_SUICIDE_IMPLANTS) &&
    entity.isActive() &&
    entity.getCrew() != null;
```

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+     * <ul>
+     *   <li>Mek: 1 point internal structure to head</li>
+     *   <li>Aero: 1 point armor to nose</li>
+     *   <li>Vehicle: 1 point internal structure to all facings</li>
+     * </ul>
+     *
+     * @return 1 (constant per IO pg 83)
+     */
+    public static int getHostDamageFor() {
+        return 1;
```
Comment (IllianiBird) [inline megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java]: Let's set this as a constant in the same place we define DAMAGE_PER_TROOPER. It means that, should these values ever change, we don't need to hunt for it.

Once we have the constant defined, this method just returns the constant

```
public static int getHostDamageFor() {
        return HOST_DAMAGE;
}
```

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+    /**
+     * Calculate building CF damage for infantry detonating inside a building. Per IO pg 83: For every 2 suicide charges
+     * set off inside a structure, subtract 1 point from the structure's CF.
+     *
+     * @param trooperCount the number of troopers detonating
+     *
+     * @return the CF damage to the building
+     */
+    public static int getBuildingDamageFor(int trooperCount) {
+        return trooperCount / 2;
```
Comment (IllianiBird) [inline megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java]: This really isn't necessary, but we can get rid of another magic number but assigning the divisor as a constant, up with DAMAGE_PER_TROOPER. Then this just becomes...

```
public static int getBuildingDamageFor(int trooperCount) {
        return trooperCount / BUILDING_DAMAGE_DIVSOR;
}
```

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+     *
+     * @return a description of the detonation effects
+     */
+    private static String getDetonationDescription(Entity entity) {
+        if (entity.isConventionalInfantry()) {
+            Infantry infantry = (Infantry) entity;
+            int maxTroopers = infantry.getShootingStrength();
+            int maxDamage = getDamageFor(maxTroopers);
+            return Messages.getString("SuicideImplantsAttackAction.detonateInfantry", maxDamage);
+        } else if (entity instanceof BattleArmor) {
```
Comment (IllianiBird) [inline megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java:242]: entity.isBattleArmor()

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+            return Messages.getString("SuicideImplantsAttackAction.detonateBA");
+        } else if (entity instanceof Mek) {
```
Comment (IllianiBird) [inline megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java]: entity.isMek()

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java [hunk 16]
```diff
@@ -0,0 +1,255 @@
...
+            return Messages.getString("SuicideImplantsAttackAction.detonateMek");
+        } else if (entity instanceof Aero) {
```
Comment (IllianiBird) [inline megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java:246]: This can stay as `entity instanceof Aero` != `entity.isAero`

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java [hunk 17]
```diff
@@ -0,0 +1,255 @@
...
+            return Messages.getString("SuicideImplantsAttackAction.detonateAero");
+        } else if (entity instanceof Tank) {
```
Comment (IllianiBird) [inline megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java]: entity.isVehicle()

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java [hunk 18]
```diff
@@ -0,0 +1,255 @@
...
+     * @return the maximum trooper count, or 1 for non-infantry units
+     */
+    public static int getMaxTroopersFor(Entity entity) {
+        if (entity.isConventionalInfantry()) {
+            return ((Infantry) entity).getShootingStrength();
+        } else if (entity instanceof BattleArmor battleArmor) {
+            return battleArmor.getShootingStrength();
+        }
+        // Meks, Aeros, Vehicles - fixed at 1 (the pilot/crew)
+        return 1;
```
Comment (IllianiBird) [inline megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java:269]: Is this correct? Meks, Aero-type units (such as DropShips), and Vehicles can all have multiple crew. What happens if a WarShip crewmember detonates a suicide implant?

File: megamek/src/megamek/server/totalWarfare/TWDamageManager.java [hunk 19]
```diff
@@ -302,6 +304,18 @@ public Vector<Report> damageEntity(Entity entity, HitData hit, int damage, boole
...
+              && hasSuicideImplants
+              && !damageType.equals(DamageType.SUICIDE_IMPLANT_REACTION);
```
Comment (IllianiBird) [inline megamek/src/megamek/server/totalWarfare/TWDamageManager.java:312]: Just clarifying that this instance of chained conditionals is fine due to how few clauses it has and the simplicity of each clause.

**Review by IllianiBird — COMMENTED** [review]
I do have a single question that I'd need answered before I can approve this PR. It's the comment related to crew sizes. If that's a non-issue, then consider this PR approved and merge at will.

Everything else is just a polish pass. Action if you want, or ignore, it's all a 'nice to have' not a 'must have'.

Great work. :)

File: megamek/src/megamek/server/totalWarfare/TWDamageManager.java (hunk 19 above)
Comment (HammerGS) [reply megamek/src/megamek/server/totalWarfare/TWDamageManager.java:312]: Fixed with https://github.com/MegaMek/megamek/pull/7837/commits/d597b43a3279522905f2a9fa3c6afc2c2836762c

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java (hunk 18 above)
Comment (HammerGS) [reply megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java:269]: Fixed with https://github.com/MegaMek/megamek/pull/7837/commits/d597b43a3279522905f2a9fa3c6afc2c2836762c made sure they are except for know add comments around not being implemented

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java (hunk 17 above)
Comment (HammerGS) [reply megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java]: Fixed with https://github.com/MegaMek/megamek/pull/7837/commits/d597b43a3279522905f2a9fa3c6afc2c2836762c

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java (hunk 16 above)
Comment (HammerGS) [reply megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java:246]: Fixed with https://github.com/MegaMek/megamek/pull/7837/commits/d597b43a3279522905f2a9fa3c6afc2c2836762c

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java (hunk 15 above)
Comment (HammerGS) [reply megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java]: Fixed with https://github.com/MegaMek/megamek/pull/7837/commits/d597b43a3279522905f2a9fa3c6afc2c2836762c

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java (hunk 14 above)
Comment (HammerGS) [reply megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java:242]: Fixed with https://github.com/MegaMek/megamek/pull/7837/commits/d597b43a3279522905f2a9fa3c6afc2c2836762c

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java (hunk 13 above)
Comment (HammerGS) [reply megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java]: Fixed with https://github.com/MegaMek/megamek/pull/7837/commits/d597b43a3279522905f2a9fa3c6afc2c2836762c

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java (hunk 12 above)
Comment (HammerGS) [reply megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java]: Fixed with https://github.com/MegaMek/megamek/pull/7837/commits/d597b43a3279522905f2a9fa3c6afc2c2836762c

File: megamek/src/megamek/client/ui/panels/phaseDisplay/FiringDisplay.java (hunk 11 above)
Comment (HammerGS) [reply megamek/src/megamek/client/ui/panels/phaseDisplay/FiringDisplay.java]: Fixed with https://github.com/MegaMek/megamek/commit/d597b43a3279522905f2a9fa3c6afc2c2836762c

File: megamek/src/megamek/client/ui/dialogs/phaseDisplay/SuicideImplantsDialog.java (hunk 10 above)
Comment (HammerGS) [reply megamek/src/megamek/client/ui/dialogs/phaseDisplay/SuicideImplantsDialog.java]: Fixed with https://github.com/MegaMek/megamek/commit/d597b43a3279522905f2a9fa3c6afc2c2836762c

File: megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java (hunk 14 above)
Comment (HammerGS) [reply megamek/src/megamek/common/actions/SuicideImplantsAttackAction.java:242]: Fixed with https://github.com/MegaMek/megamek/commit/d597b43a3279522905f2a9fa3c6afc2c2836762c

File: megamek/src/megamek/client/ui/dialogs/phaseDisplay/SuicideImplantsDialog.java (hunk 9 above)
Comment (HammerGS) [reply megamek/src/megamek/client/ui/dialogs/phaseDisplay/SuicideImplantsDialog.java:67]: Fixed with https://github.com/MegaMek/megamek/commit/d597b43a3279522905f2a9fa3c6afc2c2836762c
